### PR TITLE
Huge update 

### PR DESCRIPTION
**Pull request by micronAMZN**
Automation functionality added to tutorial dialog and tutorial step. Rigid Body PhysX tutorial has a skeleton for automation, but the automate functions are all stubs. The tutorial can be played automatically at 2 seconds per step.

Moved show() to end of update_widget() which seems to fix the issue where a double highlight would appear on the first step.

Highlight widget is now parented to app window or highlight item so it follows the app if the window is moved. Highlight can now be occluded by other applications/windows, it doesn’t sit on top of the entire desktop.

Highlight rect now based on parent widget rect. paintEvent() updated so that the widget and highlight rect scale with the parent widget.

Special handling added for highlighting the viewport so that the highlight isn’t clipped by the viewport.

Had to add closeEvent() for dialog to clean up the highlight_widget on close.

Highlight width and color changes. Highlight is a rhodamine red border of 4 pixels. Up for discussion, but looking to do something higher contrast (and more visible) than cyan.

Removed the Demo Tutorial since the Highlighting UI Objects tutorial covers all the Demo covered and more.

Text improvements to all the available tutorials.

Renamed tutorial files and tiles for consistency.

Reordered tutorials in interface.

Did a bunch of other stuff I’m forgetting as well. I think I made some changes to how steps were counted a couple months back.

Signed-off-by: Mike Cronin <58789750+micronAMZN@users.noreply.github.com>

### CODE REVIEW

**Comment by micronAMZN** [comment]
Pic of highlight change.

newhighlight

File: Editor/Scripts/demo_tutorial.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 SPDX-License-Identifier: Apache-2.0 OR MIT
 """
 
-from PySide2.QtWidgets import QMenuBar
+from PySide2.QtWidgets import QMenuBar, QMainWindow
```
Comment (cgalvan) [inline Editor/Scripts/demo_tutorial.py]: Nitpick: It doesn't look like this `QMainWindow` import gets used

File: Editor/Scripts/interactivetutorials_dialog.py [hunk 2]
```diff
@@ -122,6 +146,7 @@ def __init__(self, parent=None):
         self.tutorial_widget = QWidget(self)
         self.tutorial_layout = QVBoxLayout()
 
+        #self.highlight_layout = QStackedLayout()
```
Comment (cgalvan) [inline Editor/Scripts/interactivetutorials_dialog.py]: Nit: Can this line be removed or is for future improvements?

File: Editor/Scripts/interactivetutorials_dialog.py [hunk 3]
```diff
@@ -266,10 +313,24 @@ def load_previous_step(self):
                 self.current_step_index -= 1
                 self.load_step(prev_step)
 
+    def enable_autoplay(self):
+        self.current_tutorial.set_autoplay(True)
+        self.back_button.hide()
+        step_counter = 1
+        while( step_counter <= self.current_tutorial_num_steps):
+            self.load_next_step()
+            QCoreApplication.processEvents()
```
Comment (cgalvan) [inline Editor/Scripts/interactivetutorials_dialog.py:321]: Do we need this `QCoreApplication.processEvents()` call?

File: Editor/Scripts/rigid_body_tutorial.py [hunk 4]
```diff
@@ -5,71 +5,94 @@
...
+#import azlmbr.editor as editor
+#import azlmbr.legacy.general as general
+#import azlmbr.entity as entity
+#import azlmbr.math as math
+#import azlmbr.prefab as prefab
+#from azlmbr.entity import EntityId
+#from editor_python_test_tools.utils import TestHelper as helper
+#import azlmbr.asset as asset
+#from PySide2.QtWidgets import QMenuBar
+#from PySide2.QtWidgets import QDialog
```
Comment (cgalvan) [inline Editor/Scripts/rigid_body_tutorial.py]: Can these be removed?

File: Editor/Scripts/rigid_body_tutorial.py [hunk 5]
```diff
@@ -5,71 +5,94 @@
...
+#    enter_game_mode            = ("Entered game mode",                       "Failed to enter game mode")
+#    exit_game_mode             = ("Exited game mode",                        "Couldn't exit game mode")
+# fmt: on
+
+#class RigidBody:
+#    gravity_enabled = False
+#    linear_velocity = 10
+#    angular_velocity = 10
+#    mass = 0.0
+
```
Comment (cgalvan) [inline Editor/Scripts/rigid_body_tutorial.py]: Can these be removed as well?

File: Editor/Scripts/highlight_ui_objects_tutorial.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    pass
+
+from tutorial import Tutorial, TutorialStep
+
+class HighlightUIObjectsTutorial(Tutorial):
+    def __init__(self):
+        super(HighlightUIObjectsTutorial, self).__init__()
+
+        self.title = "Highlighting UI Objects"
+        #self.is_automated = False
```
Comment (cgalvan) [inline Editor/Scripts/highlight_ui_objects_tutorial.py]: Can this line be removed?

**Review by cgalvan — APPROVED** [review]

File: Editor/Scripts/demo_tutorial.py (hunk 1 above)
Comment (micronAMZN) [reply Editor/Scripts/demo_tutorial.py]: fixed

File: Editor/Scripts/interactivetutorials_dialog.py (hunk 2 above)
Comment (micronAMZN) [reply Editor/Scripts/interactivetutorials_dialog.py]: fixed

File: Editor/Scripts/interactivetutorials_dialog.py (hunk 3 above)
Comment (micronAMZN) [reply Editor/Scripts/interactivetutorials_dialog.py:321]: Yes, I don't remember specifically why, but the automated tutorial does crash without it.

File: Editor/Scripts/rigid_body_tutorial.py (hunk 4 above)
Comment (micronAMZN) [reply Editor/Scripts/rigid_body_tutorial.py]: fixed

File: Editor/Scripts/rigid_body_tutorial.py (hunk 5 above)
Comment (micronAMZN) [reply Editor/Scripts/rigid_body_tutorial.py]: fixed

File: Editor/Scripts/highlight_ui_objects_tutorial.py (hunk 6 above)
Comment (micronAMZN) [reply Editor/Scripts/highlight_ui_objects_tutorial.py]: fixed

**Pull request merged**